$OpenBSD: patch-configure_d_config_os_progs,v 1.6 2020/09/04 20:06:11 sthen Exp $

Net-SNMP tries to use -Wl,-no-undefined to detect missing linkage at build
time (rather than dying at runtime if referenced functions are unavailable -
this happened with some AES-related functions in an earlier release where
libm wasn't linked to the library correctly).

When trying to link the shared libraries, this fails on OpenBSD with missing
libc-related symbols. I think this maybe because OpenBSD doesn't record
libc as needed in shared libraries, presumably as a result of this
-no-undefined is tripped.

Neuter the autoconf check here to avoid this problem.

https://github.com/net-snmp/net-snmp/issues/178

Index: configure.d/config_os_progs
--- configure.d/config_os_progs.orig
+++ configure.d/config_os_progs
@@ -207,21 +207,7 @@ case x$target_os in
 esac
 
 
-#
-#    Whether the linker supports the flag -Wl,-no-undefined.
-#
-
-if ! echo "$CFLAGS" | grep -q -- -fsanitize=; then
-saved_LDFLAGS=$LDFLAGS
-AC_MSG_CHECKING([whether the linker supports -Wl,-no-undefined])
-LDFLAGS="$saved_LDFLAGS -Wl,-no-undefined"
-AC_LINK_IFELSE([AC_LANG_PROGRAM([],[])],
-               [AC_MSG_RESULT([yes]); dnl
-                LD_NO_UNDEFINED=-Wl,-no-undefined],
-               [AC_MSG_RESULT([no])])
-LDFLAGS="$saved_LDFLAGS"
 AC_SUBST(LD_NO_UNDEFINED)
-fi
 
 
 #
